Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1233,6 +1236,25 @@ static ExceptionOr<void> checkPopoverValidity(Element& element, PopoverVisibilit
     return { };
 }
 
+void HTMLElement::queuePopoverToggleEventTask(PopoverVisibilityState oldState, PopoverVisibilityState newState)
+{
+    if (auto queuedEventData = popoverData()->queuedToggleEventData())
+        oldState = (*queuedEventData).oldState;
+    popoverData()->setQueuedToggleEventData({ oldState, newState });
+    queueTaskKeepingThisNodeAlive(TaskSource::DOMManipulation, [this, newState] {
+        auto queuedEventData = popoverData()->queuedToggleEventData();
+        if (!queuedEventData)
+            return;
+        if ((*queuedEventData).newState != newState)
+            return;
+        popoverData()->clearQueuedToggleEventData();
+        auto visibilityStateToString = [](PopoverVisibilityState state) {
```

Comment:
Since this has narrow scope, it doesn’t need such a long name. Could just call it `string` even.